src/App.tsx

```diff
@@ -15,36 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = 'default',
+}
+
+interface FilterParams {
+  sortType: SortType;
+  reversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortType, reversed }: FilterParams,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.DEFAULT);
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortType,
+    reversed,
+  });
+  const reset = () => {
+    setSortType(SortType.DEFAULT);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.ALPHABET,
```

Comment:
It's important to indicate to the user which action is currently active. In this case, consider adding a visual cue to the 'Sort alphabetically' button to denote it's the current sort method when 'sortType' is 'SortType.ALPHABET'. [UX]